apps/wallet-mobile/src/features/RampOnOff/useCases/StartRampOnOff/ResultExchangeScreen/ResultExchange.tsx

```diff
@@ -0,0 +1,142 @@
+import {RouteProp, useRoute} from '@react-navigation/native'
+import * as React from 'react'
+import {Image, StyleSheet, TouchableOpacity, View} from 'react-native'
+
+import image from '../../../../../assets/img/banxa.png'
+import {Button, Icon, Spacer, Text, useModal} from '../../../../../components'
+import {useTheme} from '../../../../../theme'
+import {Theme} from '../../../../../theme/types'
+import {useHideBottomTabBar} from '../../../../../yoroi-wallets/hooks'
+import DescribeAction from '../../../common/DescribeAction'
+import {useNavigateTo} from '../../../common/navigation'
+import {useStrings} from '../../../common/strings'
+import WalletAssetImage from '../../../common/WalletAssetImage'
+import ContentResult from './ContentResult'
+
+export type ParamsResult = {
+  coin: string
+  coinAmount: number
+  fiat: number
+  fiatAmount: number
+}
+
+const ResultExchangeScreen = () => {
+  const strings = useStrings()
+  useHideBottomTabBar()
+
+  const navigateTo = useNavigateTo()
+  const route = useRoute<RouteProp<{params: ParamsResult}>>()
+
+  const {theme} = useTheme()
+
+  const styles = React.useMemo(() => getStyles({theme: theme}), [theme])
+  const params = route.params
+  const {coin, coinAmount, fiat, fiatAmount} = params ?? {}
```

Comment:
params are unsafe, please leverage useParams + typeguard for the expected types.